html_css/floats.md

```diff
@@ -0,0 +1,39 @@
+
+Let's say we want a second paragraph to *not* wrap around the image, but to be displayed not only beneath the first paragraph but also entirely beneath the image. We can then use `clear: left;` to allow this second paragraph to clear the image that has been floated left. `clear` can be assigned left, right, or both to clear items which have been floated to the left, right, or in both directions.
+
+### Problems with float
+While `float` was first used to wrap text around an image, as developers realised that you can float *anything* the property quickly became a commonplace tool for creating creating webpage layouts. This led to problems such as the one we're about to discuss'.
+
+Say you want to take your image, and your first paragraph, and put them into a box that will contain those two elements. If we attempt to `float: left;` the image so that the first paragraph is on its right, and then `clear` the second paragraph past the box container - it won't work! This happens because the float was taken out of the normal flow of the document, and even though the second paragraph comes after the first paragraph, it will no longer clear. The most modern solution to this problem is to use `display: flow-root`, which creates a miniture layout within the page, inside of which everything is contained.
+
+Problems such as this one used to require various hacks to work around, and are part of the reason why using floats to create layouts is now considered a [legacy layout method](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Legacy_Layout_Methods), and using CSS Grid or Flexbox is preferred.
+
+### Shapes
+Wrapping text around images with float is relatively straightforward, but you can also have fun with other shapes by using `shapes-outside`. This property allows text to wrap around various shapes, which you can read about in [this article by css-tricks](https://css-tricks.com/almanac/properties/s/shape-outside/). 
+
+While this may seem like it's not that different, things get really crazy when we have a look at [Chrome's CSS Shapes Editor Extention](https://chrome.google.com/webstore/detail/css-shapes-editor/nenndldnbcncjmeacmnondmkkfedmgmp?hl=en-US). Whoa! While by no means necessary, this is a fascinating interactive tool that lets you create and adjust the shapes with an interactive editor.
+
+### Assignment
+<div class="lesson-content__panel" markdown="1">
+1. For some really quick practice using `float`, MDN created [three quick exercises](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Floats_skills) you can use to acquire familiarity with its proper use.
+</div>
+
+### Additional Resources
+This section contains helpful links to other content. It isn't required, so consider it supplemental for if you need to dive deeper into something
+* [For a more in-depth look at float](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Floats) MDN has some great documentation on `float`.
```

Comment:
* [For a more in-depth look at float](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Floats) MDN has some great documentation.